modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -431,6 +431,24 @@
     [_ _ expr]
     (with-temporal-type (hsql/call bigquery-fn expr) :timestamp)))
 
+(defmethod sql.qp/->honeysql [:bigquery-cloud-sdk :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [timestamp    (partial hsql/call :timestamp)
+        datetime     (partial hsql/call :datetime)
+        clause       (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? clause "timestamp")]
+    (when (and timestamptz? source-timezone)
+      (throw (ex-info (tru "`timestamp` columns shouldn''t have a `source timezone`")
+                      {:type            qp.error-type/invalid-query
+                       :target-timezone target-timezone
+                       :source-timezone source-timezone})))
+    (let [source-timezone (or source-timezone (qp.timezone/results-timezone-id))]
+      (cond-> clause
+        (and (not timestamptz?) source-timezone)
+        (timestamp source-timezone)
+        true
+        (datetime target-timezone)))))
```

Comment:
Also, the result should be wrapped in a call to `with-temporal-type`. Or you can use `(->temporal-type :datetime expr)`

Reply:
I'm surprised this passed tests too. We should have a test for nesting convertTimezone functions.

e.g. `convertTimezone(convertTimezone([Created At], "America/Los_Angeles"), "Europe/Paris")` should return the correct result.

Reply:
nice. I went with with-temporal-type ee2179f55d6448b1153ef4b4b5ce7f73b0a6ec19